frontend/src/metabase/questions/components/QuestionMoveToast/QuestionMoveToast.tsx

```diff
@@ -15,31 +18,24 @@ interface QuestionMoveToastProps {
   question: Question;
 }
 
-const getMessage = (question: Question, collectionLink: JSX.Element) => {
-  const type = question.type();
-
-  if (type === "question") {
-    return jt`Question moved to ${collectionLink}`;
-  }
-
-  if (type === "model") {
-    return jt`Model moved to ${collectionLink}`;
-  }
-
-  if (type === "metric") {
-    return jt`Metric moved to ${collectionLink}`;
-  }
-
-  throw new Error(`Unknown question.type(): ${type}`);
-};
-
 function QuestionMoveToast({ collectionId, question }: QuestionMoveToastProps) {
   const id = coerceCollectionId(collectionId);
-  const collectionLink = <CollectionLink key="collection-link" id={id} />;
+  const type = question.type();
+  const { data: collection } = useGetCollectionQuery({ id });
+  const collectionLink = collection ? (
+    <CollectionLink key="collection-link" to={Urls.collection(collection)}>
+      {collection.name}
+    </CollectionLink>
+  ) : (
+    <Fragment key="fragment" />
+  );
+
   return (
     <ToastRoot>
       <StyledIcon name="collection" />
-      {getMessage(question, collectionLink)}
+      {type === "question" && jt`Question moved to ${collectionLink}`}
+      {type === "model" && jt`Model moved to ${collectionLink}`}
+      {type === "metric" && jt`Metric moved to ${collectionLink}`}
```

Comment:
Should we handle unknown question types here? Either by showing something like `Item moved to ...` or by throwing (so tests would catch this?

This is one of those things where we might add a type and forget about adding a new message here.

Or we could use `ts-pattern` here so the types catch it.

Reply:
> Should we handle unknown question types here? Either by showing something like Item moved to ... or by throwing (so tests would catch this?

Currently it's not possible to have different card type.

> Or we could use `ts-pattern` here so the types catch it.

If we're going to do this then we should do it everywhere because there's lots of places with similar card type based conditions. It'd deserve a dedicated PR.

Reply:
Use `ts-pattern` with `exhaustive` for a compile-type check

Reply:
No reason not to do it here with ts-pattern now fwiw

Reply:
2b63f7f395365fae8deb3783d8d36b111ed7860e